packages/frontend/src/components/TestSubstep/TestMultiRowResultModal.tsx

```diff
@@ -0,0 +1,169 @@
+
+import { Column, DataRow, processData } from './utils'
+
+interface TestMultiRowResultModalProps {
+  isOpen: boolean
+  onClose: () => void
+  variables: Variable[] | null
+  step: IStep
+}
+
+const TableHeader = ({ columns }: { columns: Column[] }) => (
+  <Thead
+    position="sticky"
+    top={0}
+    bg="var(--chakra-colors-primary-50)"
+    zIndex={1}
+  >
+    <Tr>
+      <Th borderRightWidth="1px" borderColor={BORDER_COLOR.DEFAULT} />
+      {columns?.map((c, colIndex) => (
+        <Th
+          key={c.key}
+          borderColor={BORDER_COLOR.DEFAULT}
+          borderRightWidth={colIndex === columns.length - 1 ? 0 : '1px'}
+        >
+          <Text
+            fontSize={FONT_SIZE.DEFAULT}
+            overflow="hidden"
+            whiteSpace="nowrap"
+            textOverflow="ellipsis"
+            maxW="100%"
+            textStyle="subhead-2"
+            userSelect="none"
+            textTransform="none"
+          >
+            {c.label}
+          </Text>
+        </Th>
+      ))}
+    </Tr>
+  </Thead>
+)
+
+const TableRow = ({
+  row,
+  index,
+  columns,
+}: {
+  row: DataRow
+  index: number
+  columns: Column[]
+}) => (
+  <Tr
+    backgroundColor={index % 2 ? ROW_COLOR.EVEN : ROW_COLOR.ODD}
+    borderColor={BORDER_COLOR.DEFAULT}
+  >
+    <Td
+      borderColor={BORDER_COLOR.DEFAULT}
+      borderRightWidth="1px"
+      fontSize={FONT_SIZE.SMALL}
+    >
+      {index + 1}
+    </Td>
+    {columns?.map((c, colIndex) => (
+      <Td
+        key={c.key}
+        borderColor={BORDER_COLOR.DEFAULT}
+        borderRightWidth={colIndex === columns.length - 1 ? 0 : '1px'}
+        fontSize={FONT_SIZE.DEFAULT}
+      >
+        {row.data[c.key] ?? ''}
+      </Td>
+    ))}
+  </Tr>
+)
+
+export default function TestMultiRowResultModal(
+  props: TestMultiRowResultModalProps,
+) {
+  const { isOpen, onClose, variables, step } = props
+
+  const isTilesStep = useMemo(() => step.appKey === 'tiles', [step.appKey])
+
+  const { rowsFound, dataRows, columns } = useMemo(
+    () => processData(variables, isTilesStep),
+    [variables, isTilesStep],
+  )
+
+  if (!variables) {
+    return null
+  }
+
+  return (
+    <Modal
+      isOpen={isOpen}
+      onClose={onClose}
+      size="6xl"
+      motionPreset="none"
+      isCentered
+    >
+      <ModalOverlay />
+      <ModalContent maxH="80vh" overflow="hidden" borderRadius="lg">
+        <ModalHeader
+          position="sticky"
+          top={0}
+          bg="white"
+          zIndex={1}
+          borderBottom="1px solid"
+          borderColor="base.divider.medium"
+        >
+          <Flex direction="column">
+            List of row(s) found
+            <Text textStyle="body-2" color="base.content.medium">
+              {rowsFound} row(s)
+            </Text>
+          </Flex>
+          <ModalCloseButton />
+        </ModalHeader>
+        <ModalBody>
+          <TableContainer
+            my={2}
+            maxH="calc(80vh - 100px)"
```

Comment:
```suggestion
            maxH="calc(75vh - 100px)"
```

cus right now, when there are many rows, i think the bottom margin a bit off
<img width="1301" alt="image" src="https://github.com/user-attachments/assets/b2478d84-0b42-4fa1-a840-072bb8ae909e" />